core/server/api_container/server/startosis_engine/kurtosis_instruction/upload_files/upload_files.go

```diff
@@ -188,14 +189,15 @@ func (builtin *UploadFilesCapabilities) TryResolveWith(instructionsAreEqual bool
 
 	// From here the instructions are equal
 	// If the hash of the files don't match, the instruction needs to be re-run
-	if !slices.Equal(filesArtifactMd5s, other.FilesArtifactMd5s) {
+	logrus.Warnf("Comparing file hash: '%s' and '%s'", filesArtifactMd5s[0], other.FilesArtifactMd5s[0])
+	if len(filesArtifactMd5s) != 1 || len(other.FilesArtifactMd5s) != 1 || !bytes.Equal(filesArtifactMd5s[0], other.FilesArtifactMd5s[0]) {
```

Comment:
for curiosity: what does len equal 1 mean in this context?

Reply:
Yeah, it's a bit of a hack here, but we know that `filesArtifactMd5s` has only one element (b/c it's assembled inside this object), and so to check for equality with `other.FilesArtifactMd5s` we just need to check that `len(other.FilesArtifactMd5s) == 1` AND the single elements of both arrays are equal. I'll see if I can rework this a little. Worst case I'll just add a comment here